### PR TITLE
BorderControl: Make height consistent with other controls

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Related:

- https://github.com/WordPress/gutenberg/issues/40893

## What?

Makes `BorderControl` height consistent with other controls by default i.e. `30px`. Also, updates `BorderBoxControl` to match. 

A new prop (`__next36pxDefaultSize`)  is introduced to allow opting into the future default of `36px`. 

## Why?

Border controls within the editor are currently inconsistent.

## How?

- Sets `BorderControl` height based on new `__next36pxDefaultSize` prop
- Adjusts inner component styles based on above prop
- Updates Storybook examples to include new prop for toggling default height
- Plumbs `__next36pxDefaultSize` prop through border control components as required

## Testing Instructions

1. Edit a post, add a group block, and select it.
2. In the Inspector Controls sidebar, compare the border and border radius controls. They should both be `30px` tall.
    _(Border color & focus styles will be addressed separately)_
3. Expand the `BorderBoxControl` and confirm the split border view still looks ok
4. Start up the Storybook and visit the [`BorderControl`](http://localhost:50240/?path=/story/components-experimental-bordercontrol--default) and [`BorderBoxControl`](http://localhost:50240/?path=/story/components-experimental-borderboxcontrol--default) component pages
    - `npm run storybook:dev`
    - Toggle the `__next36pxDefaultSize` control and check component renders ok 

## Screenshots or screencast 

| Before | After |
|---|---|
| Screen Shot 2022-05-09 at 4 00 04 pm | Screen Shot 2022-05-09 at 3 59 15 pm |

### CODE REVIEW

**Comment by aaronrobertshaw** [comment]
@ciampo Do you have any thoughts on better ways to approach setting the height for this control given there should be a shift to `36px` height controls soon?

I'm not sure how easy it will be to target the inner components/elements if we were to try and style it only from the editor side for now.

**Review by andrewserong — APPROVED** [review]
Hard-coding this to `30px` height in the interim until we get to a standardized `36px` sounds good to me! It looks a tiny bit tight when using a dotted border (which is improved when rebasing on `trunk` where the control has a darker border color), but otherwise I think the consistent height with the other controls makes this a worthwhile change. Thanks for adding the inline comments so it's clear what should be changed in a follow-up when we're ready to go to `36px`!

| This PR | This PR rebased on trunk so it has a darker border |
| --- | --- |
| image | image |

✅ Tested well in Storybook
✅ Looks good to me when adjusting the Group block's border controls in the editor

LGTM!

**Comment by aaronrobertshaw** [comment]
Thanks for the review @andrewserong 👍 

> It looks a tiny bit tight when using a dotted border

I agree about this appearing a bit tight and that is despite reducing the size of the color swatch already. 

A different option was to increase the size of the border-radius control to match the height of the `BorderControl` but that only increased consistency within the Border panel so I opted against that.

> I think the consistent height with the other controls makes this a worthwhile change

I'm inclined to agree that the improved height consistency is the better option in the short term. 

Before looking to merge this, I'll see if I can steal another pixel of space by reducing the gap between the color swatch and its border e.g. the dotted border you mentioned. 

In addition, I'll see if I can add something like the `__next36pxDefaultSize` prop from `InputControl` here and adjust styles accordingly. I'm not sure that will be any cleaner, now or when removing the `30px` height though.

**Comment by aaronrobertshaw** [comment]
@andrewserong I've reworked the approach in this PR so the control can opt into the future default height of `36px` more easily.

It should also make it pretty clear what code should be removed when the `30px` height is to be deprecated. It also lays the groundwork for supporting multiple heights out of the box.

If you have the time for another review, I'd like to get your thoughts on this. If you think the extra changes are too much we can always revert the last commit.

P.S. This has also been rebased to include the updated border colors.


https://user-images.githubusercontent.com/60436221/167562117-8bb41c16-fb67-4291-afdb-458cf106ebb7.mp4

**Review by andrewserong — APPROVED** [review]
> It should also make it pretty clear what code should be removed when the 30px height is to be deprecated. It also lays the groundwork for supporting multiple heights out of the box.

Nice work @aaronrobertshaw! This is still testing well, the smaller size looks better now it's rebased on trunk and with the adjusted spacing, and it's very cool being able to switch that flag in Storybook and see it immediately render correctly with the new height 👍

I think I usually lean toward avoiding unneeded complexity, but in this case, I think it's worth including the additional changes for a couple of reasons:

* The added complexity is pretty straightforward (it's a boolean flag that's clearly documented)
* Adding in the ability to do this now saves someone the trouble further down the track of having to look at the component and work out how best to update the height again, and furthers the efforts in #39397

Also it looks like it follows the conventions @mirka used in https://github.com/WordPress/gutenberg/issues/39397, so this LGTM! 🎉

**Comment by ciampo** [comment]
> @ciampo Do you have any thoughts on better ways to approach setting the height for this control given there should be a shift to `36px` height controls soon?

The current approach looks good to me.

More details on the plans to increase the default size to `36px` in #39397 (which also contains links to the PR that @mirka has recently worked on) 

> I'm not sure how easy it will be to target the inner components/elements if we were to try and style it only from the editor side for now.

I'm not sure I get this point

**Review by ciampo — APPROVED** [review]
Changes LGTM 🚀 

I only have 2 thoughts (not blocking, though):

- In components which have a few sub-components (like `BorderBoxControl`), I wonder if using the internal component's context system could be a better way to propagate the `__next36pxDefaultSize` prop instead of "forwarding" it explicitly as a prop on each component
- In the styles, I see a lot of hardcoded, arbitrary numbers — I wonder if it made sense to use the `space` utility more often and/or if it made sense to show the logic behind these numbers (e.g.:
    ```js
  const targetHeight = 36;
  const borderWidth = 1;
  const vertcalPadding = 2;
  const innerHeight = targetHeight - 2 * ( borderWidth + vertcalPadding );
    ```

Again, more of a reflection (and definitely not blocking)!

**Comment by aaronrobertshaw** [comment]
Thanks for the sanity check and the quick review @ciampo 👍 

> I only have 2 thoughts

Appreciate the suggestions as always. I'll look to adopt them in a follow-up to this PR.

Given the component height does feed into some other styling tweaks being made to the border control components, I'll merge this one now.

**Comment by aaronrobertshaw** [comment]
> > I'm not sure how easy it will be to target the inner components/elements if we were to try and style it only from the editor side for now.
> 
> I'm not sure I get this point

I was questioning whether I should be adding what will become an outdated default height to the component and therefore whether styling the component's height via block editor styles would be better. I didn't think it would really work let alone not be brittle.

With the latest approach, the question was a bit obsolete.

**Comment by andrewserong** [comment]
While reviewing, I noticed the spacing between the border radius' `UnitControl` and the range control is slightly narrower than the `BorderControl`'s. I've opened up a tiny PR in #40958 to try to get this more consistent with the extra room in `BorderControl` 🙂

**Comment by aaronrobertshaw** [comment]
Thanks, it was on my list to clean up. I'll review it shortly.

**Pull request merged**
